extensions/wikia/AchievementsII/services/AchAwardingService.class.php

```diff
@@ -466,26 +486,39 @@ private function processAllInTrack() {
 				}
 			}
 		}
+	}
 
-
-		// BADGE_LOVE
-		if(empty($this->mCounters[BADGE_LOVE])) {
+	private function processInTrackLove() {
+		//BADGE_LOVE
+		if (empty($this->mCounters[BADGE_LOVE])) {
 			$this->mCounters[BADGE_LOVE][COUNTERS_COUNTER] = 1;
 		} else {
-			if($this->mCounters[BADGE_LOVE][COUNTERS_DATE] == date('Y-m-d')) {
+			if ($this->mCounters[BADGE_LOVE][COUNTERS_DATE] == date('Y-m-d')) {
 				// ignore
-			} else if($this->mCounters[BADGE_LOVE][COUNTERS_DATE] == date('Y-m-d', strtotime('-1 day'))) {
+			} else if ($this->mCounters[BADGE_LOVE][COUNTERS_DATE] == date('Y-m-d', strtotime('-1 day'))) {
 				$this->mCounters[BADGE_LOVE][COUNTERS_COUNTER]++;
 			} else {
 				$this->mCounters[BADGE_LOVE][COUNTERS_COUNTER] = 1;
 			}
 		}
 		$this->mCounters[BADGE_LOVE][COUNTERS_DATE] = date('Y-m-d');
+	}
 
-		wfProfileOut(__METHOD__);
+	private function incrementCounterBy($badge_type_id, $value) {
+		$this->mCounters = is_array($this->mCounters) ? $this->mCounters : $this->mUserCountersService->getCounters();
```

Comment:
if mCounters is not an array then mUserCountersService is not set either - so what's the point of this conditional check?